iot/api/CloudIotMqttExample/Program.cs

```diff
@@ -0,0 +1,645 @@
+            // required to be in the format below. Note that this is not the same as the device registry's
+            // Cloud Pub/Sub topic.
+            string mqttTopic = $"/devices/{deviceId}/{sub_topic}";
+
+            SetupMqttTopics(client, deviceId);
+
+            //Publish num_messages mesages to the MQTT bridge once per second.
+            for (var i = 1; i <= numMsgs; ++i)
+            {
+                string payload = string.Format("{0}/{1}-payload-{2}", registryId, deviceId, i);
+                Console.WriteLine("Publishing {0} message {1}/{2}: '{3}'\n", messageType, i, numMsgs, payload);
+                var BinaryData = Encoding.Unicode.GetBytes(payload);
+
+                // Refresh the connection credentials before the JWT expires.
+                long secsSinceRefresh = (DateTime.Now.Millisecond - iat.Millisecond) / 1000;
+                if (secsSinceRefresh > 60 * jwtExpiresMin)
+                {
+                    Console.WriteLine("\tRefreshing token after: {0} seconds\n", secsSinceRefresh);
+                    iat = DateTime.Now;
+                    if (algorithm == "RS256")
+                    {
+                        pass = CreateJwtRsa(projectId, privateKeyFile);
+                    }
+                    else if (algorithm == "ES256")
+                    {
+                        //TODO: needs to be implemented
+                    }
+                    else
+                    {
+                        throw new ArgumentException("Invalid algorithm {0}. Should be one of 'RS256' or 'ES256'", algorithm);
+                    }
+                    client.Disconnect();
+                    client.Connect(clientId, "unused", pass);
+                }
+
+                // Publish "payload" to the MQTT topic. qos=1 means at least once delivery. Cloud IoT Core
+                // also supports qos=0 for at most once delivery.
+                client.Publish(mqttTopic, BinaryData, MqttMsgBase.QOS_LEVEL_AT_LEAST_ONCE, false);
+
+                if (messageType.Equals("event"))
+                {
+                    // Send telemetry events every second
+                    System.Threading.Thread.Sleep(1000);
+                }
+                else
+                {
+                    // Note: Update Device state less frequently than with telemetry events
+                    System.Threading.Thread.Sleep(5000);
+                }
+            }
+
+            for (int i = 1; i < waitTime; i++)
+            {
+                Console.Write(".");
+                System.Threading.Thread.Sleep(1000);
+            }
+            Console.WriteLine();
+
+            // Disconnect the client if still connected, and finish the run.
+            if (client.IsConnected)
+            {
+                client.Disconnect();
+            }
+            Console.WriteLine("Finished loop successfully. Goodbye!");
+            return 0;
+        }
```

Comment:
This function is 141 lines long.  I don't see a natural place to break it into smaller functions; do you?

Reply:
I will break this down into seperate functions